src/cursor/cur_backup_incr.c

```diff
@@ -0,0 +1,233 @@
+/*-
+ * Copyright (c) 2014-2019 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __alloc_merge --
+ *     Merge two allocation lists.
+ */
+static void
+__alloc_merge(
+  uint64_t *a, uint64_t a_cnt, uint64_t *b, uint64_t b_cnt, uint64_t *res, uint64_t *res_cnt)
+{
+    uint64_t total;
+
+    for (total = 0; a_cnt > 0 || b_cnt > 0; ++total, res += 2) {
+        if (a_cnt > 0 && b_cnt > 0) {
+            if (a[0] <= b[0]) {
+                res[0] = a[0];
+                if (a[0] + a[1] < b[0])
+                    res[1] = a[1];
+                else {
+                    res[1] = (b[0] + b[1]) - a[0];
+                    b += 2;
+                    --b_cnt;
+                }
+                a += 2;
+                --a_cnt;
+            } else if (b[0] <= a[0]) {
+                res[0] = b[0];
+                if (b[0] + b[1] < a[0])
+                    res[1] = b[1];
+                else {
+                    res[1] = (a[0] + a[1]) - b[0];
+                    a += 2;
+                    --a_cnt;
+                }
+                b += 2;
+                --b_cnt;
+            }
+        } else if (a_cnt > 0) {
+            res[0] = a[0];
+            res[1] = a[1];
+            a += 2;
+            --a_cnt;
+        } else if (b_cnt > 0) {
+            res[0] = b[0];
+            res[1] = b[1];
+            b += 2;
+            --b_cnt;
+        }
+    }
+    *res_cnt = total;
```

Comment:
I think this all goes away -- let's talk through the plan for how we're getting/merging lists of block modifications (unless you've already decided what you want to do).

Reply:
This function is from your original work. It is not being used at the moment because the incremental cursor is only returning "entire file" information.